chaos-monkey-spring-boot/src/main/java/de/codecentric/spring/boot/chaos/monkey/assaults/CpuAssault.java

```diff
@@ -0,0 +1,83 @@
+package de.codecentric.spring.boot.chaos.monkey.assaults;
+
+import com.sun.management.OperatingSystemMXBean;
+import de.codecentric.spring.boot.chaos.monkey.component.MetricEventPublisher;
+import de.codecentric.spring.boot.chaos.monkey.component.MetricType;
+import de.codecentric.spring.boot.chaos.monkey.configuration.ChaosMonkeySettings;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CpuAssault implements ChaosMonkeyRuntimeAssault {
+  private static final Logger Logger = LoggerFactory.getLogger(CpuAssault.class);
+
+  private final ChaosMonkeySettings settings;
+
+  private final MetricEventPublisher metricEventPublisher;
+  private final OperatingSystemMXBean os;
+
+  public CpuAssault(
+      OperatingSystemMXBean os,
+      ChaosMonkeySettings settings,
+      MetricEventPublisher metricEventPublisher) {
+    this.os = os;
+    this.settings = settings;
+    this.metricEventPublisher = metricEventPublisher;
+  }
+
+  @Override
+  public boolean isActive() {
+    return settings.getAssaultProperties().isCpuActive();
+  }
+
+  @Override
+  public void attack() {
+    Logger.info("Chaos Monkey - cpu assault");
+
+    // metrics
+    if (metricEventPublisher != null) {
+      metricEventPublisher.publishMetricEvent(MetricType.CPU_ASSAULT);
+    }
+    double load = settings.getAssaultProperties().getCpuLoadTargetFraction();
+
+    List<Thread> threads = new ArrayList<>();
+    for (int num = 0; os.getProcessCpuLoad() < load; num++) {
+      Thread thread =
+          new Thread(
+              () -> {
+                while (!Thread.interrupted()) ;
```

Comment:
My understanding is that spawning new threads actually consumes more memory. I wonder if the CPU assault is instead using up memory, then forcing the application to perform garbage collection, which reflects as high CPU load.

I wonder if the thread should be doing some CPU intensive work as well or if this alone is sufficient.

Reply:
Threads do consume memory, though not enough to trigger excessive garbage collection in my testing. Nonetheless, reducing the number of threads necessary seems like a good idea, so I added some fibonacci computation, as suggested in the original issue.